app/src/main/java/com/example/taskyy/ui/viewmodels/RegisterViewModel.kt

```diff
@@ -21,21 +24,24 @@ class RegisterViewModel @Inject constructor(
             is RegisterEvent.OnNameChanged -> state = state.copy(name = event.name)
             is RegisterEvent.OnEmailChanged -> state = state.copy(email = event.email)
             is RegisterEvent.OnPasswordChanged -> state = state.copy(password = event.password)
-            is RegisterEvent.OnGetStartedClick -> register()
+            is RegisterEvent.OnGetStartedClick -> register(context = event.applicationContext)
             is RegisterEvent.OnTogglePasswordVisibility -> state = state.copy(isPasswordVisible = event.isPasswordVisible)
-                   }
+        }
     }
 
-    private fun register() {
-        authRepository.registerUser(name = state.name, password = state.password, email = state.email)
+    private fun register(context: Context) {
+        val user = User(name = state.name, password = state.password, email = state.email)
+        val dao = TaskyyDatabase.getDatabase(context = context).userDao()
+        authRepository.registerUser(user, dao)
     }
 }
 
-    data class RegisterState(
+data class RegisterState(
         var email: String = "",
         var password: String = "",
         var name: String = "",
         var isEmailValid: Boolean = false,
-        var isPasswordVisible: Boolean = false
+        var isPasswordVisible: Boolean = false,
+        val context: Context? = null
```

Comment:
If this was the activity context, you'd cause a memory leak here. The whole point of the VM is to outlive the lifecycle of the activity, but when keeping a reference to the activity in the VM, the garbage collector won't be able to free up this reference when the activity is destroyed (like on a screen rotation).
So you would effectively keep a reference to an activity that isn't used anymore. I know you only passed the app context here, but this is very dangerous